src/setting.rs

```diff
@@ -17,11 +17,12 @@
 //!
 //! The exact format still has to be reverse-engineered.
 
-use binrw::{binrw, NullString};
+use binrw::{binrw, io::Cursor, BinWrite, Endian, NullString, WriteOptions};
 
 #[binrw]
 #[derive(Debug, PartialEq)]
 #[brw(little)]
+#[bw(import(no_checksum: bool))]
```

Comment:
since this is basically a hack, I'd propose to make give it a indicative name and assign an appropriate default
```suggestion
#[bw(import(_no_checksum: bool = false))]
```

Reply:
I don't think the default argument is possible. And I don't understanrd the purpose of the underscore.

Reply:
My current understanding is that the `no_checksum` argument is basically a hack so we can serialize the `Setting` struct once without a checksum and once with. This is a workaround because of lacking support to do this "properly" with binrw. As such, it is an implementation detail. If I just want to serialize the `Setting` struct, I don't want to worry about this internal hack. As such, I propose to make the `no_checksum` attribute optional by using an appropriate default argument and indicating that its private (by prefixing it with `_` per convention). 

defaults args are supposed to be possible per documentation (see the example code and single sentence https://docs.rs/binrw/latest/binrw/attribute/read/index.html#named-arguments).

Reply:
Ah, but then import needs a struct, not a tuple.

Reply:
Ah, with named arguments a struct is generated and we'd have to reference its name (which is an implementation detail of binrw) in the `write_options` implementation :(
It also interferes with the `missing_docs` clippy rule.

And the leading underscore prevents any unused-variable warnings, doesn't it?

Reply:
> Ah, with named arguments a struct is generated and we'd have to reference its name (which is an implementation detail of binrw) in the write_options implementation :(

Mhmm. that sounds like an issue worth raising on the binrw bugtracker. 

> And the leading underscore prevents any unused-variable warnings, doesn't it?

true, I forgot about that convention. 

I really wish there was a solution that didn't expose the double-serialization workaround in the public interface, but I guess there is none.

Reply:
> I really wish there was a solution that didn't expose the double-serialization workaround in the public interface, but I guess there is none.

Yeah. Maybe we can come up with a better implementation later on.